src/_pytest/python.py

```diff
@@ -1181,19 +1181,25 @@ def _idval(
         if hook_id:
             return hook_id
 
+    modified_val = None  # empty strings are valid as val
     if isinstance(val, STRING_TYPES):
-        return _ascii_escaped_by_config(val, config)
+        modified_val = _ascii_escaped_by_config(val, config)
     elif val is None or isinstance(val, (float, int, bool)):
-        return str(val)
+        modified_val = str(val)
     elif isinstance(val, REGEX_TYPE):
-        return ascii_escaped(val.pattern)
+        modified_val = ascii_escaped(val.pattern)
     elif isinstance(val, enum.Enum):
-        return str(val)
+        modified_val = str(val)
     elif isinstance(getattr(val, "__name__", None), str):
         # name of a class, function, module, etc.
-        name = getattr(val, "__name__")  # type: str
-        return name
-    return str(argname) + str(idx)
+        modified_val = getattr(val, "__name__")
+
+    # val does not always enter the isinstance checks due to its type
+    # when it does we will check the string length returned and use auto generation of ids when over 100 chars
+    # on types which do not match isinstance checks pytest uses auto generate of ids automatically anyway
```

Comment:
I suggest to reword this a bit more succinctly:

```suggestion
    # If val is not of one of the types we support above, or if it ends up too long to
    # be useful, use a generated name based on the argument name and index instead.
```